src/odemis/dataio/csv.py

```diff
@@ -45,159 +45,260 @@ def export(filename, data):
-        if acq_type != model.MD_AT_EK:
-            logging.warning("Data seems to be EK, but acq_type is %s", acq_type)
+        _export_temporal_spectrum_data(acq_type, data, filename)
 
-        spectrum_range, unit_c = spectrum.get_spectrum_range(data)
-        if unit_c == "m":
-            spectrum_range = [s * 1e9 for s in spectrum_range]
-            unit_c = "nm"
+    # Export Angular Spectrum Data
+    elif dims == "AC" and data.ndim == 2:
+        logging.debug("Exporting angular spectrum data to CSV")
+        _export_angular_spectrum_data(acq_type, data, filename)
 
-        angle_range, unit_a = spectrum.get_angle_range(data)
-        if unit_a == "rad":
-            unit_a = "°"
-            angle_range = [math.degrees(theta) for theta in angle_range]  # Convert radians to degrees
-            assert len(angle_range) == len(data)
+    # Export AR Data
+    elif acq_type == model.MD_AT_AR:
+        logging.debug("Exporting AR data to CSV")
+        _export_ar_data(data, filename)
 
-        headers = ["angle(" + unit_a + ")\\wavelength(" + unit_c + ")"] + spectrum_range
-        rows = [(t,) + tuple(d) for t, d in zip(angle_range, data)]
+    else:
+        raise ValueError(f"Unknown acquisition type {acq_type} of data (dims = {dims}) to be exported as CSV")
 
-        with open(filename, 'w') as fd:
-            csv_writer = csv.writer(fd)
-            csv_writer.writerow(headers)
-            csv_writer.writerows(rows)
 
+def _export_spectrum_data(acq_type, data, filename):
+    """
+    This will export the spectrum data to a CSV file using the type of microscope,
+    the acquired data and filename title in its creation.
+    :param acq_type: data.metadata.get(model.MD_ACQ_TYPE, None)
+    :param data: (model.DataArray)
+    :param filename: (unicode)
+    """
+
+    if acq_type != model.MD_AT_SPECTRUM:
+        logging.warning("Data seems to be spectrum, but acq_type is %s", acq_type)
+
+    spectrum_range, unit = spectrum.get_spectrum_range(data)
+    if unit == "m":
+        # turn range to nm
+        spectrum_range = [s * 1e9 for s in spectrum_range]
+        spectrum_tuples = [(s, d) for s, d in zip(spectrum_range, data)]
+        headers = ['# wavelength (nm)', 'intensity']
     else:
-        raise ValueError(f"Unknown acquisition type {acq_type} of data (dims = {dims}) to be exported as CSV")
+        logging.info("Exporting spectrum without wavelength information")
+        spectrum_tuples = data.reshape(data.shape[0], 1)
+        headers = ['# intensity']
+
+    with open(filename, 'w', newline='') as fd:
+        csv_writer = csv.writer(fd)
+        csv_writer.writerow(headers)
+        csv_writer.writerows(spectrum_tuples)
+
+    fd.close()
+
+
+def _export_chronogram_data(data, filename):
+    """
+    This will export the chronogram data to a CSV file using the acquired data,
+    and filename title in its creation.
+    :param data:
+    :param filename:
+    :return:
```

Comment:
there is no return, so you can remove this line. This is the same for your other functions